### PR TITLE
Wait for started up a web server after rackup on the "test:templates"

### PR DESCRIPTION
**Pull request by sh19910711**
Sometimes a connection to web server fails on the "test:templates" rake task. This pull request is to fix it by adding a "wait" task.

Failed case: https://travis-ci.org/rails/web-console/jobs/75395250

### CODE REVIEW

**Comment by sh19910711** [comment]
@gsamokovarov I just tried to fix the case. Could you please review the Rakefile?

Thanks!

**Comment by gsamokovarov** [comment]
Can we also investigate if we can run this only on only one travis worker and not all of the Ruby ones?

**Comment by sh19910711** [comment]
http://docs.travis-ci.com/user/customizing-the-build/#Explicity-Including-Builds

I guess this feature would be useful for that. I will investigate it :-)

**Comment by gsamokovarov** [comment]
Man, those Rakefiles are pretty big. We may need to think of extracting some of their code in PORO or [`Rake::Task`](http://ruby-doc.org/stdlib-2.2.2/libdoc/rake/rdoc/Rake/Task.html) objects.

File: Rakefile [hunk 1]
```diff
@@ -25,16 +25,24 @@ namespace :test do
...
     pid_file    = Pathname(Dir.tmpdir).join("web_console.#{SecureRandom.uuid}.pid")
-    server_port = 29292
-    rackup_opts = "-p #{server_port}"
-    test_runner = "http://localhost:#{server_port}/html/spec_runner.html"
+    runner_uri  = URI.parse("http://localhost:29292/html/spec_runner.html")
+    rackup_opts = "-p #{runner_uri.port}"
     test_result = nil
 
+    def need_to_wait?(uri)
+      begin
```
Comment (gsamokovarov) [inline Rakefile]: No need for the `begin` here. The body of the method acts as implicit begin block, so you can do:

```ruby
def some_method
  call_an_action_that_may_raise_an_error
rescue SomeErrorType
  handle_it_here
end
```

File: Rakefile [hunk 2]
```diff
@@ -25,16 +25,24 @@ namespace :test do
...
+        Net::HTTP.start(uri.host, uri.port) { |http| http.get(uri.path) }
+        return false
```
Comment (gsamokovarov) [inline Rakefile]: No need for the return here. Also on line 42.

File: Rakefile [hunk 3]
```diff
@@ -47,8 +55,12 @@ namespace :test do
       Dir.chdir(work_dir) { system "bundle exec rackup #{rackup_opts}" }
     end
 
+    task :wait do
+      5.times { if need_to_wait?(runner_uri) then sleep(1) else break end }
```
Comment (gsamokovarov) [inline Rakefile]: Hm, can't we actually put the retry logic in `need_to_wait`? See [retry](http://ruby-doc.com/docs/ProgrammingRuby/html/tut_expressions.html#UM).

**Comment by gsamokovarov** [comment]
Thanks!

**Comment by sh19910711** [comment]
Thanks for merging and reviewing :)

**Pull request merged**